python/graphstorm/dataloading/metadata.py

```diff
@@ -113,39 +113,43 @@ def __init__(self,
 
         if nfeat_dims is not None:
             assert isinstance(nfeat_dims, dict) and all(isinstance(key, str) for key in \
-                nfeat_dims.keys()) and all(isinstance(val, dict) for val in \
-                nfeat_dims.values()) and all(isinstance(val_key, str) for val in \
-                nfeat_dims.values() for val_key in val.keys()) and all( \
-                isinstance(val_val, list) for val in nfeat_dims.values() for val_val in \
-                val.values()), ('The node feature dimensions should be a dictionary, whose '
-                    'keys are node type strings, and values are dictionaries whose keys are '
-                    'node feature name strings, and values are lists of feature dimensions, '
-                    f'but got {nfeat_dims}.')
-            assert all(ntype in self._ntypes for ntype in nfeat_dims.keys()), 'Some node ' + \
-                f'types in node feature dimensions: {nfeat_dims} are not in the node type ' + \
-                f'list: {ntypes}.'
+                nfeat_dims.keys()), ('The node feature dimensions should be a dictionary, ' \
+                    f'whose keys are node type strings, but got {nfeat_dims}.')
+            assert all(isinstance(val, dict) for val in nfeat_dims.values()) and \
+                all(isinstance(val_key, str) for val in nfeat_dims.values() for val_key \
+                in val.keys()) and all(isinstance(val_val, list) for val in nfeat_dims.values() \
+                for val_val in val.values()), ('The node feature dimension dictionary\'s ' \
+                    'values must have a string as the dictionary key, and a list of feature ' \
+                    f'dimensions as the value, but got {nfeat_dims}.')
+            # need to make sure the input argument format is correct first, and then can test
+            # the contents
+            assert all(ntype in self._ntypes for ntype in nfeat_dims.keys()), ('Some node ' \
+                f'types in node feature dimensions: {nfeat_dims} are not in the node type ' \
+                f'list: {ntypes}.')
         self._nfeat_dims = nfeat_dims
 
         if efeat_dims is not None:
             assert isinstance(efeat_dims, dict) and all(isinstance(key, tuple) for key in \
-                efeat_dims.keys()) and all(len(key)==3 for key in efeat_dims.keys()) and \
-                all(isinstance(val, dict) for val in efeat_dims.values()) and all( \
+                efeat_dims.keys()) and all(len(key)==3 for key in efeat_dims.keys()), ( \
```

Comment:
Can you further break the assertion?
If this check fails, users have to figure out whether it is because 1/ The edge feature dimensions should be a dictionary or 2/ keys are canonical edge types.